.github/workflows/master.yml

```diff
@@ -128,11 +127,21 @@ jobs:
           java-version: ${{ matrix.java }}
           cache: 'maven'
           check-latest: false
-      - name: Build and test Flink with maven w/o linters
-        run: |       
-          TEST_MODULES="externals/kyuubi-flink-sql-engine,integration-tests/kyuubi-flink-it"
+      - name: Build Flink with maven w/o linters
+        run: |
+          ENGINE_MODULE=externals/kyuubi-flink-sql-engine
+          IT_MODULE=integration-tests/kyuubi-flink-it
+          TEST_MODULES="${ENGINE_MODULE},${IT_MODULE}"
           ./build/mvn ${MVN_OPT} -pl ${TEST_MODULES} -Pflink-${{ matrix.flink }} ${{ matrix.flink-archive }} -am  clean install -DskipTests
+      - name: Test Flink
+        if: matrix.flink-archive == ''
+        run: |
           ./build/mvn ${MVN_OPT} -pl ${TEST_MODULES} -Pflink-${{ matrix.flink }} ${{ matrix.flink-archive }} test
+      - name: Cross-version test Flink
+        if: matrix.flink-archive != ''
+        run: |
+          IT_FLINK=`echo "${{ matrix.flink-archive }}" | grep -E 'flink\-([0-9]+\.[0-9]+.[0-9]+)\-bin' -o | grep -E '[0-9]+\.[0-9]+' -o`
+          ./build/mvn ${MVN_OPT} -pl ${IT_MODULE} -Pflink-${IT_FLINK} ${{ matrix.flink-archive }} test
```

Comment:
The cross-version verification composes of:
- flink-sql-engine compiled against flink-1.15
- flink-1.14 binary tgz
- flink-1.14 mini cluster
- flink-1.14 it test suites

Reply:
cc @deadwind4 FYI